onnx/defs/sequence/defs.cc

```diff
@@ -0,0 +1,235 @@
+        .TypeConstraint(
+            "S",
+            OpSchema::all_tensor_sequence_types(),
+            "Constrain output types to any tensor type.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // TODO: assign datatype here.
+        }));
+
+static const char* SequenceConstruct_ver11_doc = R"DOC(
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SequenceConstruct,
+    11,
+    OpSchema()
+        .SetDoc(SequenceConstruct_ver11_doc)
+        .Input(
+            0,
+            "inputs",
+            "List of tensors.",
+            "T",
+            OpSchema::Variadic)
+        .Output(
+            0,
+            "output_sequence",
+            "Sequence enclosing the input tensors.",
+            "S")
+        .TypeConstraint(
+            "T",
+            OpSchema::all_tensor_types(),
+            "Constrain input types to any tensor type.")
+        .TypeConstraint(
+            "S",
+            OpSchema::all_tensor_sequence_types(),
+            "Constrain output types to any tensor type.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+          auto numInputs = ctx.getNumInputs();
+          if (numInputs < 1 ||
+              !hasNInputShapes(ctx, static_cast<int>(numInputs))) {
+            return;
+          }
+
+          // Should we set shape info if input shapes are all known and are the same?
+        }));
+
+static const char* SequenceInsert_ver11_doc = R"DOC(
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SequenceInsert,
+    11,
+    OpSchema()
+        .SetDoc(SequenceInsert_ver11_doc)
+        .Input(
+            0,
+            "input_sequence",
+            "Input sequence.",
+            "S")
+        .Input(
+            1,
+            "tensor",
+            "Input tensor to be inserted into the input sequence.",
+            "T")
+        .Input(
+            2,
+            "position",
+            "Position in the sequence where the new tensor is inserted. "
+            "It is optional and default is to insert to the back of the sequence. "
+            "It must be a scalar(tensor of empty shape).",
+            "I",
+            OpSchema::Optional)
+        .Output(
+            0,
+            "output_sequence",
+            "Output sequence that contains the inserted tensor at given position.",
+            "S")
+        .TypeConstraint(
+            "T",
+            OpSchema::all_tensor_types(),
+            "Constrain to any tensor type.")
+        .TypeConstraint(
+            "S",
+            OpSchema::all_tensor_sequence_types(),
+            "Constrain to any tensor type.")
+        .TypeConstraint(
+            "I",
+            {"tensor(int32)", "tensor(int64)"},
+            "Constrain position to integral tensor. It must be a scalar(tensor of empty shape).")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
```

Comment:
As above. We need a different utility function for sequences, which are different from tensors.